src/sugar3/graphics/progressicon.py

```diff
@@ -14,86 +14,171 @@
+        if self._direction == 'vertical':  #vertical direction, bottom to top 
             y_ = self._stroke.get_height()
             height *= self._progress * -1
-        else:
+        else: 
             rtl_direction = \
-                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL
-            if rtl_direction:  # horizontal direction, right to left
-                x_ = self._stroke.get_width()
+                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL '''Sets text direction from right to left'''
+            if rtl_direction: # horizontal direction, right to left
+                x_ = self._stroke.get_width() 
                 width *= self._progress * -1
-            else:  # horizontal direction, left to right
+            else: # horizontal direction, left to right
                 width *= self._progress
 
         cr.rectangle(x_, y_, width, height)
-        cr.clip()
+        cr.clip() 
         cr.set_source_surface(self._fill, 0, 0)
-        cr.paint()
+        cr.paint() 
 
         # Paint the stroke over the fill.
-        cr.reset_clip()
+        cr.reset_clip() 
         cr.set_source_surface(self._stroke, 0, 0)
         cr.paint()
 
     def do_get_preferred_width(self):
-        width = self._stroke.get_width()
-        return (width, width)
+        width = self._stroke.get_width() 
+        return (width, width) 
+        '''
+        Returns:
+        
+            Width for progressicon. Width will be set to _stroke.get_width()
+       
+        Example:
+            from gi.repository import Gtk
+            class Patch(Gtk.DrawingArea):
+            def __init__(self):
+            super(Patch,self).__init__()
+
+            def do_get_preferred_width(self):
+            print("example")
+            return 100, 100
+        '''
 
     def do_get_preferred_height(self):
         height = self._stroke.get_height()
-        return (height, height)
-
-    def update(self, progress):
-        self._progress = progress
-        self.queue_draw()
+        return (height, height) '''Returns tuple'''
+        '''
+        Returns:
+        
+            Width for progressicon. Height will be set to _stroke.get_height()
+        
+        Example:
+            from gi.repository import Gtk
+            class Patch(Gtk.DrawingArea):
+            def __init__(self):
+            super(Patch,self).__init__()
+
+            def do_get_preferred_height(self):
+            print("example")
+            return (100, 100)
+        '''
+
+    def update(self, progress): '''updates progressicon with progress value '''
+        self._progress = progress '''Set progressbar value'''
+        self.queue_draw() 
+        '''
+        Example:
+            def update(self, iterable):
+            for item in iterable:
+            self.items_list.append(item)
+            
+        Returns:
+            Updates progressicon with progress value 
+         '''
```

Comment:
This is not a perfectly clear example on the usage of this function, because the function arguments are not the same. Here, progress is not a list, rather a value which tells how much of the process is completed. An example can for this can be as simple as "update(0.9)".